components/sup/src/manager/service/hooks.rs

```diff
@@ -20,6 +19,25 @@ use std::{self,
 
 static LOGKEY: &str = "HK";
 
+#[derive(Debug)]
+pub struct CompleteHookOutput {
+    stdout:      Option<String>,
+    stderr:      Option<String>,
+    exit_status: ExitStatus,
+}
+
+impl CompleteHookOutput {
+    fn new(hook_output: &HookOutput, exit_status: ExitStatus) -> Self {
+        Self { stdout: hook_output.stdout_str(),
+               stderr: hook_output.stderr_str(),
+               exit_status }
+    }
+
+    pub fn get_exit_status(&self) -> &ExitStatus { &self.exit_status }
```

Comment:
`ExitStatus` implements `Copy`, so we may as well avoid the unnecessary references. Also, [the `get_` prefix is generally omitted in idiomatic rust](https://rust-lang-nursery.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter), giving:
```suggestion
    pub fn exit_status(&self) -> ExitStatus { self.exit_status }
```